src/Glpi/Asset/AssetDefinition.php

```diff
@@ -744,27 +750,46 @@ public function getFieldOrder(): array
      * @param bool $check_values
      * @return bool
      */
-    private function validateCapacityArray(mixed $capacities, bool $check_values = true): bool
+    private function validateCapacityArray(mixed &$capacities, bool $check_values = true): bool
     {
+        if ($capacities instanceof \stdClass) {
+            $capacities = (array)$capacities;
+        }
```

Comment:
This should not be required if you decode it using `, associative: true`.

Reply:
If we use associative in json encode, the swhole decode will eb an array. But what I want is an array of objects